src/items/reward.ts

```diff
@@ -384,50 +413,89 @@ export class TerastallizeModifierType extends PokemonModifierType {
   }
 
   get name(): string {
-    return i18next.t("modifierType:ModifierType.TerastallizeModifierType.name", {
+    return i18next.t("modifierType:ModifierType.TerastallizeReward.name", {
       teraType: i18next.t(`pokemonInfo:Type.${PokemonType[this.teraType]}`),
     });
   }
 
   getDescription(): string {
-    return i18next.t("modifierType:ModifierType.TerastallizeModifierType.description", {
+    return i18next.t("modifierType:ModifierType.TerastallizeReward.description", {
       teraType: i18next.t(`pokemonInfo:Type.${PokemonType[this.teraType]}`),
     });
   }
 
   getPregenArgs(): any[] {
     return [this.teraType];
   }
+
+  /**
+   * Checks if {@linkcode TerrastalizeConsumable} should be applied
+   * @param playerPokemon The {@linkcode PlayerPokemon} that consumes the item
+   * @returns `true` if the {@linkcode TerrastalizeConsumable} should be applied
+   */
+  shouldApply({ pokemon }: PokemonRewardParams): boolean {
+    return (
+      [pokemon?.species.speciesId, pokemon?.fusionSpecies?.speciesId].filter(
+        s => s === SpeciesId.TERAPAGOS || s === SpeciesId.OGERPON || s === SpeciesId.SHEDINJA,
+      ).length === 0
```

Comment:
```suggestion
      pokemon.teraType !== this.teraType &&
      ![SpeciesId.SHEDINJA, SpeciesId.OGERPON, SpeciesId.TERAPAGOS].some(s => hasSpecies(s))
```

Doesn't have to be changed now but I want to point it out as something that should be changed